### PR TITLE
fix(cli): handle invalid inline base64 source map gracefully to prevent panic

### PR DESCRIPTION
**Pull request by yazan-abdalrahman**
#21989

### CODE REVIEW

**Comment by yazan-abdalrahman** [comment]
before
image

after
image

**Review by lucacasonato — COMMENTED** [review]
Can you explain what this code change is meant to do?

**Comment by yazan-abdalrahman** [comment]
> Can you explain what this code change is meant to do?

While debugging, its failed on  **" .expect()"** in **cli/util/text_encoding.rs method source_map_from_code**,
so, I debugged the parameter **'code'**

I saw "code" value =  **[cli\util\text_encoding.rs:11:3] &code = "\"use strict\";\r\n\r\nthrow new Error(\"Hello world!\");\r\n//# sourceMappingURL=data:application/json;base64,{"**
When running a script, end with '.js'.

but for running a script, end with '.ts'.
value = **[cli\util\text_encoding.rs:11:3] &code = "\"use strict\";\nthrow new Error(\"Hello world!\"); //# sourceMappingURL=data:application/json;base64,{\n//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImZpbGU6Ly8vRDoveWFkL2Rlbm8vbWFpbi5qcyJdLCJzb3VyY2VzQ29udGVudCI6WyJcInVzZSBzdHJpY3RcIjtcclxuXHJcbnRocm93IG5ldyBFcnJvcihcIkhlbGxvIHdvcmxkIVwiKTtcclxuLy8jIHNvdXJjZU1hcHBpbmdVUkw9ZGF0YTphcHBsaWNhdGlvbi9qc29uO2Jhc2U2NCx7Il0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBO0FBRUEsTUFBTSxJQUFJLE1BQU0saUJBQ2hCLG1EQUFtRCJ9"**

So, I saw the difference, it's related to changing the flow at this method 
**cli/module_loader.rs load_prepared_module_or_defer_emit**

image
When a file type is TypeScript, it returns the source as part of Result 
**DeferredEmit {
    specifier: &'a ModuleSpecifier,
    media_type: MediaType,
    source: &'a Arc<str>,
  }**

but when a file type is JavaScript its doesn't return source, 
it's return
 ModuleCodeStringSource {
  pub code: ModuleCodeString,
  pub found_url: ModuleSpecifier,
  pub media_type: MediaType,
}
image


so, I changed match to add source and it's work

**Comment by bartlomieju** [comment]
@dsherret can you take a look here? Looks like this is touching the same bits as you did in https://github.com/denoland/deno/pull/24094

File: cli/module_loader.rs [hunk 1]
```diff
@@ -684,15 +684,15 @@ impl<TGraphContainer: ModuleGraphContainer>
...
-          | MediaType::Unknown
+          MediaType::Unknown
           | MediaType::Cjs
           | MediaType::Mjs
           | MediaType::Json => source.clone().into(),
           MediaType::Dts | MediaType::Dcts | MediaType::Dmts => {
             Default::default()
           }
-          MediaType::TypeScript
+          MediaType::JavaScript
```
Comment (dsherret) [inline cli/module_loader.rs:694]: I don't think we should be emitting JavaScript. The fix is likely somewhere else.

Also, adding a test to demonstrate what the issue is here would be useful. See the `tests/specs` folder.

Comment (dsherret) [reply cli/module_loader.rs:694]: Actually, the changes in https://github.com/denoland/deno/pull/24094/files#diff-b8f09652408b91ba94406ad379632895db8b42df251d8d8c5060f70f2e986f98 will fix this original issue.

**Pull request closed without merging**